src/engine/strat_engine/tests/util.rs

```diff
@@ -9,43 +9,72 @@ use devicemapper::{DM, DmFlags, DevId};
 use nix::mount::{MNT_DETACH, umount2};
 use mnt::get_submounts;
 
+
+mod cleanup_errors {
+    error_chain!{}
+}
+
+use self::cleanup_errors::*;
+
+
 /// Attempt to remove all device mapper devices which match the stratis naming convention.
 /// FIXME: Current implementation complicated by https://bugzilla.redhat.com/show_bug.cgi?id=1506287
-fn dm_stratis_devices_remove() {
+fn dm_stratis_devices_remove() -> Result<()> {
 
     let dm = DM::new().unwrap();
 
+    let mut remain = Vec::new();
     loop {
         let mut progress_made = false;
-        for d in dm.list_devices()
-                .unwrap()
+
+        let device_list = dm.list_devices()
+            .chain_err(|| "Unable to retrieve dm devices")?;
+        for d in device_list
                 .iter()
                 .filter(|d| format!("{}", d.0.as_ref()).starts_with("stratis-1")) {
-            progress_made |= dm.device_remove(&DevId::Name(&d.0), DmFlags::empty())
-                .is_ok();
+
+            match dm.device_remove(&DevId::Name(&d.0), DmFlags::empty()) {
+                Ok(_) => progress_made = true,
+                Err(_) => remain.push(format!("{}", d.0.as_ref())),
+            }
         }
 
         if !progress_made {
+            if remain.len() != 0 {
+                bail!("We were unable to remove all stratis device mapper devices {:?}",
+                      remain);
+            }
             break;
+        } else {
+            remain.clear();
```

Comment:
Think this qualifies as premature optimization. Why not just initialize the vector in the loop?